fabric-biomes-v1/src/testmod/java/net/fabricmc/fabric/test/biome/FabricBiomeTest.java

```diff
@@ -16,26 +16,52 @@
 
 package net.fabricmc.fabric.test.biome;
 
-import net.minecraft.world.biome.DefaultBiomeCreator;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.registry.BuiltinRegistries;
 import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
 import net.minecraft.world.biome.Biome;
-import net.minecraft.world.biome.Biomes;
+import net.minecraft.world.biome.BuiltInBiomes;
+import net.minecraft.world.biome.DefaultBiomeCreator;
 
 import net.fabricmc.api.ModInitializer;
 import net.fabricmc.fabric.api.biomes.v1.NetherBiomes;
 import net.fabricmc.fabric.api.biomes.v1.OverworldBiomes;
+import net.fabricmc.fabric.api.biomes.v1.OverworldClimate;
 
+/**
+ * <b>NOTES FOR TESTING:</b>
+ * When running with this test-mod, also test this when running a dedicated server since there
+ * are significant differences between server + client and how they sync biomes.
+ *
+ * <p>Ingame, you can use <code>/locatebiome</code> since we use nether- and end-biomes in the overworld,
+ * and vice-versa, making them easy to find to verify the injection worked.
+ *
+ * <p>If you don't find a biome right away, teleport far away (~10000 blocks) from spawn and try again.
+ */
 public class FabricBiomeTest implements ModInitializer {
 	public static final String MOD_ID = "fabric-biome-api-v1-testmod";
 
-	@Override public void onInitialize() {
-		Biome biome = Registry.register(BuiltinRegistries.BIOME, new Identifier(MOD_ID, "test_crimson_forest"), DefaultBiomeCreator.createCrimsonForest());
-		NetherBiomes.addNetherBiome(Biomes.BEACH, new Biome.MixedNoisePoint(0.0F, 0.5F, 0.0F, 0.0F, 0.1F));
-		NetherBiomes.addNetherBiome(biome, new Biome.MixedNoisePoint(0.0F, 0.5F, 0.0F, 0.0F, 0.275F));
+	private static final RegistryKey<Biome> TEST_CRIMSON_FOREST = RegistryKey.of(Registry.BIOME_KEY, new Identifier(MOD_ID, "test_crimson_forest"));
+
+	private static final RegistryKey<Biome> CUSTOM_PLAINS = RegistryKey.of(Registry.BIOME_KEY, new Identifier(MOD_ID, "custom_plains"));
+
+	@Override
+	public void onInitialize() {
+		Registry.register(BuiltinRegistries.BIOME, TEST_CRIMSON_FOREST.getValue(), DefaultBiomeCreator.createCrimsonForest());
```

Comment:
I know you disagree, but I'm still gonna comment that I don't think we should use `BuiltinRegistries`, but `DynamicRegistryManager.BUILTIN`. We can easily provide a getter for that registry too.

Reply:
Paired with #1052 this is the easiest to explain and safest approach of registering worldgen objects, since it is also very easy to follow through the code (and check against Vanilla's behavior)